addons/crm/models/crm_stage.py

```diff
@@ -48,3 +48,21 @@ def default_get(self, fields):
     @api.depends('team_id')
     def _compute_team_count(self):
         self.team_count = self.env['crm.team'].search_count([])
+
+    def write(self, vals):
+        """ Since leads that are in a won stage must have their
+        probability = 100%, this override ensures that setting a stage as won
+        will set all the leads in that stage to probability = 100%.
+        Inversely, if a won stage is not marked as won anymore, the lead
+        probability should be recomputed based on automated probability.
+        Note: If a user sets a stage as won and changes his mind right after,
+        the manual probability will be lost in the process."""
+        res = super().write(vals)
+        if 'is_won' in vals:
+            won_leads = self.env['crm.lead'].search([('stage_id', 'in', self.ids)])
+            if won_leads and vals.get('is_won'):
```

Comment:
Can potentially be violent ... shouldn't we have a warning if somehow someone changes the flag ?